### PR TITLE
Snapshot testing

### PR DESCRIPTION
**Pull request by rachitnigam**
- The `--no-compile` option to `dios` prints out intermediate compiled
file before being compiled to C++.
- Add snapshot testing with runt.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,6 +8,8 @@ Compile code for DSPs.
 - Install [Rosette][] by running `raco pkg install rosette`.
 - Install Racket's threading library with `raco pkg install threading`
 - Install [Z3][] and [Boolector][].
+- (Testing) Install [runt][] by running `cargo install runt`. This requires
```
Comment (avanhatt) [inline README.md:11]: "for testing" ?

File: src/c-ast.rkt [hunk 2]
```diff
@@ -109,8 +109,12 @@
 
 (struct c-comment (str) #:transparent)
 
+(struct c-empty () #:transparent)
+
 (define (to-string prog [tab-size 0])
+     ;(pretty-print prog)
```
Comment (avanhatt) [inline src/c-ast.rkt:115]: Remove?

File: src/compiler.rkt [hunk 3]
```diff
@@ -44,397 +44,3 @@
                      trunc-prog
                      #:fn-map (check-transform-with-fn-map)))))
   trunc-prog)
-
```
Comment (avanhatt) [inline src/compiler.rkt:47]: yay removing these

File: Makefile [hunk 4]
```diff
@@ -1,11 +1,20 @@
-.PHONY: test build
+.PHONY: test build test-all dios dios-example-gen
 
-test:
-	raco test src/*.rkt src/backend/*.rkt
+# By default, run two jobs
```
[the remark this answers is not in the document]
Comment (rachitnigam) [reply Makefile:3]: Hm, it's the just the jobs that `make` runs in parallel. The comment is meant to change a `make` command into `make -j2` automatically.

**Comment by rachitnigam** [comment]
Comments addressed. Merge on approve.

**Pull request merged**